package/MDAnalysis/core/Selection.py

```diff
@@ -17,520 +17,540 @@
 
 """
 Atom selection Hierarchy --- :mod:`MDAnalysis.core.Selection`
-======================================================================
+=============================================================
 
 These objects are constructed and applied to the group
 
-Currently all atom arrays are handled internally as sets, but returned as AtomGroups
+In general, Parser.parse() creates a Selection object
+from a selection string.
 
+This Selection object is then passed an AtomGroup through its
+apply method to apply the Selection to the AtomGroup.
+
+This is all invisible to the user through ag.select_atoms
 """
 
+import collections
 import re
 import numpy as np
 from numpy.lib.utils import deprecate
 from Bio.KDTree import KDTree
 import warnings
+import logging
 
-from .AtomGroup import AtomGroup, Universe
 from MDAnalysis.core import flags
 from ..lib import distances
 from ..lib.mdamath import triclinic_vectors
+from ..exceptions import SelectionError, NoDataError
 
 
-class Selection(object):
-    def _apply(self, group):
-        # This is an error
-        raise NotImplementedError("No _apply function defined")
+logger = logging.getLogger(__name__)
 
-    def apply(self, group):
-        # Cache the result for future use
-        # atoms is from Universe
-        # returns AtomGroup
-
-        # make a set of all the atoms in the group
-        # XXX this should be static to all the class members
```

Comment:
To answer Oli's comment - looks like I never implemented this :(

But if git blame is correct it was written 8 yrs ago :)